File: SpinnersDialogsIntents/app/src/main/java/fei/mx/uv/spinnersdialogsintents/MainActivity.java
package fei.mx.uv.spinnersdialogsintents;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Spinner;

public class MainActivity extends AppCompatActivity {

    private String carrera;
    private String materia;
    public String[] materias = {
            "DAR",
            "Base de Datos I",
            "Base de Datos II",
            "Ingenieria de Software",
            "Redes I",
            "Redes II",
            "Redes III"
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Spinner spinner = (Spinner) findViewById(R.id.spinner2);
        ArrayAdapter<String> dataAdapter =
                new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, materias);
        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinner.setAdapter(dataAdapter);
    }

    public void onClicked(View v) {
        Spinner spn_carrera = (Spinner) findViewById(R.id.spinner);
        Spinner spn_materia = (Spinner) findViewById(R.id.spinner2);
        carrera = spn_carrera.getSelectedItem().toString();
        materia = spn_materia.getSelectedItem().toString();
        //Mostrar dialogo
        AlertDialog dialogo = new AlertDialog.Builder(MainActivity.this).create();
        dialogo.setTitle("Valores seleccionados");
        dialogo.setMessage(String.format("Los valores seleccionados son:" +
                "\nCarrera: %s\nMateria: %s\n\n¿Deseea continuar?", carrera, materia));
        dialogo.setButton(AlertDialog.BUTTON_POSITIVE, "Acepatar", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                llamarSegundaActivididad();
            }
        });
        dialogo.setButton(AlertDialog.BUTTON_NEGATIVE, "Cancelar", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
        dialogo.show();
    }

    private void llamarSegundaActivididad() {
        Intent intent = new Intent(this, SegundaActivity.class);
        intent.putExtra("val_carrera", carrera);
        intent.putExtra("val_materia", materia);
        startActivity(intent);
    }

}

File: FormularioRegistro/app/src/main/java/fei/mx/uv/formularioregistro/MainActivity.kt
package fei.mx.uv.formularioregistro

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner

class MainActivity : AppCompatActivity() {

    /**
     *  Programar sin modificar variables no imposible ;)
     * */

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        (findViewById(R.id.registroButton) as Button).setOnClickListener {
            onClickAction()
        }
    }

    fun validateTextFields(editText: EditText): Boolean {
        val b = editText.text.isEmpty()
        if (b) editText.error = "El campo no puede estar vacio!"
        return b
    }

    fun onClickAction(): Unit {
        val editTexts = listOf<View>(
                findViewById(R.id.nombresInput),
                findViewById(R.id.apellidosInput),
                findViewById(R.id.emailInput),
                findViewById(R.id.telefonoInput))
        val validated = editTexts.map { v -> validateTextFields(v as EditText) }
        if (validated.all{ b -> !b })
            createDialog(editTexts +
                    findViewById(R.id.ciudadesSpinner) +
                    findViewById(R.id.comidasSpinner))
    }

    fun createDialog(views: List<View>): Unit {
        val dialogo = AlertDialog.Builder(this@MainActivity).create()
        dialogo.setTitle("Registro")
        val valores = viewToText(views)
        val message = "Los valores ingresados son:\n" +
                "\nNombres: " + valores[0] +
                "\nApellidos: " + valores[1] +
                "\nEmail: " + valores[2] +
                "\nTelefono: " + valores[3] +
                "\nCiudad: " + valores[4] +
                "\nComida: " + valores[5] +
                "\n\nEsta seguro de querer realizar el registro?"
        dialogo.setMessage(message)
        dialogo.setButton(AlertDialog.BUTTON_POSITIVE, "Acepatar") { dialog, which -> startNewActivity(valores) }
        dialogo.setButton(AlertDialog.BUTTON_NEGATIVE, "Cancelar") { dialog, which -> dialog.dismiss() }
        dialogo.show()
    }

    fun viewToText(views: List<View>): List<String> {
        return views.map {
            when (it) {
                is EditText -> it.text
                is Spinner  -> it.selectedItem.toString()
                else -> "" }
        }.map { it.toString() }
    }

    fun startNewActivity(valores: List<String>): Unit {
        val intent = Intent(this, SegundaActivity::class.java)
        intent.putExtra("val_nombres", valores[0])
        intent.putExtra("val_apellidos", valores[1])
        intent.putExtra("val_email", valores[2])
        intent.putExtra("val_telefono", valores[3])
        intent.putExtra("val_ciudad", valores[4])
        intent.putExtra("val_comida", valores[5])
        startActivity(intent)
    }

}
